### PR TITLE
Update ruff to version 0.5.2 and workflows update

### PR DESCRIPTION
**Pull request by Smartappli**
Workflow updated

### CODE REVIEW

**Comment by qubvel** [comment]
Hi @Smartappli , thanks for the opening PR!
Let me know when its ready 🙂

**Comment by Smartappli** [comment]
> Hi @Smartappli , thanks for the opening PR! Let me know when its ready 🙂

Hi @qubvel, its ready

**Comment by qubvel** [comment]
Can you please alse update ruff in setup.py?

**Comment by qubvel** [comment]
One more note here, can you please pin timm version back? It's going to be updated in another PR, thanks.

**Comment by Smartappli** [comment]
@qubvel ready to merge

File: ruff.toml [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    "buck-out",
+    "build",
+    "dist",
+    "node_modules",
+    "site-packages",
+    "venv",
+]
+
+# Same as Black.
+line-length = 88
```
Comment (qubvel) [inline ruff.toml:32]: I would prefer a longer line, e.g. 120 chars

File: ruff.toml [hunk 2]
```diff
@@ -0,0 +1,61 @@
+# Exclude a variety of commonly ignored directories.
```
Comment (qubvel) [inline ruff.toml:1]: Can you please tell what was the motivation for adding this config?

File: requirements.txt [hunk 3]
```diff
@@ -1,4 +1,4 @@
-torchvision>=0.5.0
+torchvision>=0.10.0
```
Comment (qubvel) [inline requirements.txt:1]: Any reason to do that?

**Review by qubvel — COMMENTED** [review]
Thanks for working on this, a few more comments below. Moreover, CI fails due to changed ruff configuration

File: ruff.toml (hunk 1 above)
Comment (Smartappli) [reply ruff.toml]: done

**Comment by qubvel** [comment]
Sorry, probably there was a kind of misunderstanding, I thins exclude is necessary, but I asked about what was the motivation to add `ruff.toml`?

**Comment by qubvel** [comment]
Sorry, I cannot merge it this way because it breaks CI. I suggest moving ruff config change to a separate PR and keeping here just ruff version change to be able to pass the CI checks.

In **next** PR the config might be created and changed, then ruff formatting applied to reflect config changes.

**Comment by Smartappli** [comment]
> Sorry, I cannot merge it this way because it breaks CI. I suggest moving ruff config change to a separate PR and keeping here just ruff version change to be able to pass the CI checks.
> 
> In **next** PR the config might be created and changed, then ruff formatting applied to reflect config changes.

reverted

**Review by qubvel — APPROVED** [review]
Looks good, thanks for iterating!

**Pull request merged**